src/components/Pages/index.js

```diff
@@ -81,29 +84,36 @@ class PoolDetailPage extends React.Component {
     this.requestSent = false
   }
 
-  componentWillMount () {
-    if (Selectors.isFilterChecked()) {
-      this.props.getPools({ poolId: this.props.match.params.id })
+  fetchThePoolIfNeeded () {
+    const { match, vms, getPools } = this.props
+    let requested = false
+
+    if (!this.requestSent && !vms.getIn(['pools', match.params.id, 'vm'])) {
+      this.requestSent = requested = true
+      getPools({ poolId: match.params.id })
     }
+
+    return requested
+  }
+
+  componentWillMount () {
+    this.fetchThePoolIfNeeded()
   }
 
   componentWillUpdate () {
-    const poolInStore = this.props.vms.getIn(['pools', this.props.match.params.id, 'vm'])
-    if (!poolInStore && Selectors.isFilterChecked() && !this.requestSent) {
-      this.requestSent = true
-      this.props.getPools({ poolId: this.props.match.params.id })
-    }
+    this.fetchThePoolIfNeeded()
   }
 
   render () {
-    let { match, vms, config, requestActive } = this.props
+    const { match, vms, config } = this.props
 
     if (vms.getIn(['pools', match.params.id, 'vm'])) {
-      return (<VmDetail vm={vms.getIn(['pools', match.params.id, 'vm'])} pool={vms.getIn(['pools', match.params.id])} config={config} isPool />)
-    } else if (requestActive) {
-      return null
+      return (<VmDetail vm={vms.getIn(['pools', match.params.id, 'vm'])} pool={vms.getIn(['pools', match.params.id])} config={config} />)
     }
-    return <Redirect to='/' />
+
+    // TODO: Add handling for if the fetch runs but fails (FETCH-FAIL)
+    console.info(`PoolDetailPage: Pool id cannot be found: ${match.params.id}`)
```

Comment:
I think the logging in general should be revisited across the code base.  `logDebug()` is ok, but there are instances where I lost important call stack/location details by using it instead of `console.(log|debug|info|warn)`.

For now, I'd say it's beyond the scope of the issue this PR is addressing.